impactlab_tools/utils/configdict.py

```diff
@@ -0,0 +1,163 @@
+"""Class for representing tool configuration files
+"""
+
+import collections
+import inspect
+
+
+def gather_configtree(d):
+    """Chains nested-dicts into a connected tree of ConfigDict(s)
+
+    Parameters
+    ----------
+    d : dict
+        Cast to :py:class:`ConfigDict`. Nested dicts within are also
+        recursively cast and assigned parents, reflecting their nested
+        structure.
+
+    Returns
+    -------
+    out : ConfigDict
+
+    Examples
+    --------
+    .. code-block:: python
+
+        >>> nest = {'a': 1, 'b': {'a': 2}, 'c': 3, 'd-4': 4, 'e_5': 5, 'F': 6}
+        >>> tree = gather_configtree(nest)
+        >>> tree['b']['d-4']
```

Comment:
Perhaps first show tree['b']['a'], and then show this with an inline comment that parent keys are also available.

Reply:
Sounds good. Done.